base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+        return A
+    end
+end
+
+transpose(x::AbstractVector) = [ transpose(v) for i=1, v in x ]
+ctranspose{T}(x::AbstractVector{T}) = T[ ctranspose(v) for i=1, v in x ] #Fixme comprehension
+
+
+# From operators.jl
+
+# should probably be map(conj, transpose(x)) but map seems to assume that elements are iterable
+# ctranspose(x) = conj(x)
+
+
+# ctranspose(a::AbstractArray) = error("ctranspose not implemented for $(typeof(a)). Consider adding parentheses, e.g. A*(B*C') instead of A*B*C' to avoid explicit calculation of the transposed matrix.")
+# transpose(a::AbstractArray) = error("transpose not implemented for $(typeof(a)). Consider adding parentheses, e.g. A*(B*C.') instead of A*B*C' to avoid explicit calculation of the transposed matrix.")
+
+
+# transposed divide
+# TODO! Probably reenable when lrdiv uses Transpose
+# Ac_rdiv_B(a,b)  = ctranspose(a)/b
+# A_rdiv_Bc(a,b)  = a/ctranspose(b)
+# Ac_rdiv_Bc(a,b) = ctranspose(a)/ctranspose(b)
+# At_rdiv_B(a,b)  = transpose(a)/b
+# A_rdiv_Bt(a,b)  = a/transpose(b)
+# At_rdiv_Bt(a,b) = transpose(a)/transpose(b)
+
+# Ac_ldiv_B(a,b)  = ctranspose(a)\b
+# A_ldiv_Bc(a,b)  = a\ctranspose(b)
+# Ac_ldiv_Bc(a,b) = ctranspose(a)\ctranspose(b)
+# At_ldiv_B(a,b)  = transpose(a)\b
+# A_ldiv_Bt(a,b)  = a\transpose(b)
+# At_ldiv_Bt(a,b) = At_ldiv_B(a,transpose(b))
+# Ac_ldiv_Bt(a,b) = Ac_ldiv_B(a,transpose(b))
+
+# TODO! Remove when lrdiv uses Transpose
+Ac_rdiv_Bc(a::Number,b::Number) = conj(a)/conj(b)
+
+# """
+#     Ac_ldiv_B(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᴴ`` \\ ``B``.
+# """
+# Ac_ldiv_B
+
+# """
+#     Ac_rdiv_B(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᴴ / B``.
+# """
+# Ac_rdiv_B
+
+# """
+#     At_rdiv_Bt(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᵀ / Bᵀ``.
+# """
+# At_rdiv_Bt
+
+# """
+#     A_ldiv_Bt(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``A`` \\ ``Bᵀ``.
+# """
+# A_ldiv_Bt
+
+# """
+#     Ac_ldiv_Bc(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᴴ`` \\ ``Bᴴ``.
+# """
+# Ac_ldiv_Bc
+
+# """
+#     At_ldiv_Bt(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᵀ`` \\ ``Bᵀ``.
+# """
+# At_ldiv_Bt
+
+# """
+#     At_ldiv_B(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᵀ`` \\ ``B``.
+# """
+# At_ldiv_B
+
+# """
+#     A_ldiv_Bc(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``A`` \\ ``Bᴴ``.
+# """
+# A_ldiv_Bc
+
+# """
+#     A_rdiv_Bc(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``A / Bᴴ``.
+# """
+# A_rdiv_Bc
+
+# """
+#     A_rdiv_Bt(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``A / Bᵀ``.
+# """
+# A_rdiv_Bt
+
+# """
+#     Ac_rdiv_Bc(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᴴ / Bᴴ``.
+# """
+# Ac_rdiv_Bc
+
+# """
+#     At_rdiv_B(A, B)
+
+# For matrices or vectors ``A`` and ``B``, calculates ``Aᵀ / B``.
+# """
+# At_rdiv_B
+
+
+const transposebaselength=64
```

Comment:
Nice implementation of blocked transpose. This looks more carefully performance-tuned than my permuteddimsarrays.jl, so it will be worth being emulating someday.